agent/engine/docker_container_engine.go

```diff
@@ -168,7 +168,27 @@ var scratchCreateLock sync.Mutex
 
 // NewDockerGoClient creates a new DockerGoClient
 func NewDockerGoClient(clientFactory dockerclient.Factory, cfg *config.Config) (DockerClient, error) {
-	client, err := clientFactory.GetDefaultClient()
+	// Use 1.21 client if container metadata is enabled; otherwise use default
+	var err error
```

Comment:
We should not be switching the default.  Instead, any call that needs a specific version can request that specific version from the factory.